src/pet.js

```diff
@@ -0,0 +1,125 @@
+// CONSTRUCTOR FOR PET OBJECTS //
+const PET_IS_BORN = 0;
+const MAXIMUM_HUNGER = 10;
+const MINIMUM_FITNESS = 0;
+
+function Pet(name) {
+    this.name = name;
+    this.age = PET_IS_BORN;
+    this.hunger = MINIMUM_HUNGER;
+    this.fitness = MAXIMUM_FITNESS;
+    this.children = [];
+};
+//GETTERS//
+//ISALIVE: Checks states to determine if pet is alive//
+
+const LIFESPAN = 30;
+
+Pet.prototype = {
+    get isAlive() {
+      return this.age < LIFESPAN && this.hunger < MAXIMUM_HUNGER && this.fitness > MINIMUM_FITNESS;
+    },
+  };
+
+// PROTOTYPE METHODS FOR PET OBJECTS //
+//GROWING UP: It adds years and hunger. It reduces fitness.//
+
+const A_YEAR = 1;
+const SOFA_TIME = 3;
+const FASTING = 5;
+
+Pet.prototype.growUp = function(){
+    if (!this.isAlive) {
+        throw new Error('Your pet is no longer alive!');
+    };
+
+    this.age += A_YEAR;
+    this.fitness -= SOFA_TIME;
+    this.hunger += FASTING;
+    if(this.fitness <= MINIMUM_FITNESS) {
+        this.fitness = MINIMUM_FITNESS;
+    };
```

Comment:
This could be refactored a bit. In essence you are subtracting a value from the existing fitness, without allowing it go under a set limit. You can do this with Math.max, which will return the highest of the values passed to it:
```
    this.age += 1; // Im not sure the `A_YEAR` constant is actually necessary
    this.hunger += FASTING;
    this.fitness = Math.max(this.fitness -= SOFA_TIME, MINIMUM_FITNESS);
```